CommunicationLayer/DataPopulators/CmuPopulator.cpp

```diff
@@ -1,57 +1,38 @@
 #include "CmuPopulator.h"
-#include <QDebug>
 
-CmuPopulator::CmuPopulator(I_PacketDecoder& packetDecoder,
-                           I_BatteryData& batteryData)
-    : packetDecoder_(packetDecoder)
-    , batteryData_(batteryData)
+CmuPopulator::CmuPopulator(I_PacketDecoder& packetDecoder, I_CmuData& cmuData)
+: packetDecoder_(packetDecoder), cmuData_(cmuData)
 {
-    connect(&packetDecoder_, SIGNAL(packetDecoded(const CmuDataMessage)),
-            this, SLOT(populateData(const CmuDataMessage)));
+    connect(&packetDecoder_ SIGNAL(packetDecoded(const CmuMessage)), this, SLOT(populateData(const CmuMessage)));
 }
 
-void CmuPopulator::populateData(const CmuDataMessage message)
+void CmuPopulator::populateData(const CmuMessage message)
 {
-    switch (message.cellNumber())
-    {
-        case 0:
-            batteryData_.setMod0PcbTemperature(message.pcbTemperature());
-            batteryData_.setMod0CellTemperature(message.cellTemperature());
-            batteryData_.setMod0CellVoltages(
-                convertFloatListToDouble(message.cellVoltages()));
-            break;
-
-        case 1:
-            batteryData_.setMod1PcbTemperature(message.pcbTemperature());
-            batteryData_.setMod1CellTemperature(message.cellTemperature());
-            batteryData_.setMod1CellVoltages(
-                convertFloatListToDouble(message.cellVoltages()));
-            break;
-
-        case 2:
-            batteryData_.setMod2PcbTemperature(message.pcbTemperature());
-            batteryData_.setMod2CellTemperature(message.cellTemperature());
-            batteryData_.setMod2CellVoltages(
-                convertFloatListToDouble(message.cellVoltages()));
-            break;
-
-        case 3:
-            batteryData_.setMod3PcbTemperature(message.pcbTemperature());
-            batteryData_.setMod3CellTemperature(message.cellTemperature());
-            batteryData_.setMod3CellVoltages(
-                convertFloatListToDouble(message.cellVoltages()));
-            break;
-    }
-}
-
-QList<double> CmuPopulator::convertFloatListToDouble(QList<float> floatList)
-{
-    QList<double> doubleList;
-
-    for (int i = 0; i < floatList.length(); i++)
-    {
-        doubleList.append((double)floatList[i]);
-    }
-
-    return doubleList;
+    I_CmuUnit* cmuUnit = cmuData_.getCmuUnit(message.cmuNumber());
+    QList<unsigned short> cellVoltages = message.cellVoltages();
+    QList<unsigned short> cellTemperatures = message.cellTemperatures();
+    cmuUnit->setCell0Voltage(cellVoltage.pop_front());
+    cmuUnit->setCell1Voltage(cellVoltage.pop_front());
+    cmuUnit->setCell2Voltage(cellVoltage.pop_front());
```

Comment:
Hey @enochtsang, did some work on the data populators tonight which made me realize the way I setup the CmuData and MpptData wasn't going to work because there were multiple of them. Changed CmuData to CmuUnit and created a new CmuData that can contain a number of CmuUnits (as for how many I will add to the settings and retrieve from there if you're happy with my implementation). The only thing I'm really not too sure about is the signals, I will look into them more when I get the chance, but if it's clear to you this isn't going to work and I should be copying the data out four times in the CmuData let me know.